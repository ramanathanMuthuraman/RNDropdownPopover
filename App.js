/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import RNDropdownPopover from './RNDropdownPopover';

import {StyleSheet, SafeAreaView, View, Text} from 'react-native';

const App = () => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.wrapper}>
        <RNDropdownPopover
          buttonProps={{
            onClick: () => {},
            children: <Text>Press here to open popover!</Text>,
          }}>
          <View style={[styles.list]}>
            <Text>Option 1</Text>
            <Text>Option 2</Text>
            <Text>Option 3</Text>
            <Text>Option 4</Text>
          </View>
        </RNDropdownPopover>
        <RNDropdownPopover
          buttonProps={{
            onClick: () => {},
            children: <Text>Press here to open popover!</Text>,
          }}>
          <View style={[styles.list]}>
            <Text>Option 1</Text>
            <Text>Option 2</Text>
            <Text>Option 3</Text>
            <Text>Option 4</Text>
          </View>
        </RNDropdownPopover>
      </View>

      {/* <RNDropdownPopover />

      <RNDropdownPopover /> */}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: 20,
  },
  button: {
    marginTop: 20,
    padding: 20,
    fontSize: 36,
    color: 'black',
  },
  list: {
    padding: 20,
    width: 400,
  },
  wrapper: {
    display: 'flex',
    flexDirection: 'row',
  },
});

export default App;
