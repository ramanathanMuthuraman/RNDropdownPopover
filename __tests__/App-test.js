/**
 * @format
 */

import 'react-native';
import React from 'react';

import App from '../App';

import {render} from '@testing-library/react-native';

jest.mock('../RNDropdownPopover');

test('renders correctly', () => {
  const container = render(<App />);
  expect(container).toMatchSnapshot();
});
