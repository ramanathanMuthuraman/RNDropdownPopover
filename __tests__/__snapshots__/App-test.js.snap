// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  />
</RCTSafeAreaView>
`;
